tcp_check/tests/test_tcp_check.py

```diff
@@ -97,18 +101,153 @@ def test_response_time(aggregator):
+            id='Dual IPv4/IPv6: string hostname, IPv4 resolution, multiple_ips False',
+        ),
+        pytest.param(
+            'another-hostname',
+            common.DUAL_STACK_GETADDRINFO_IPV6,
+            [addr_tuple('ip1', socket.AF_INET6)],
+            False,
+            1,
+            id='Dual IPv4/IPv6: string hostname, IPv6 resolution, multiple_ips False',
+        ),
+        pytest.param(
+            'some-hostname',
+            common.DUAL_STACK_GETADDRINFO_IPV4,
+            [addr_tuple('ip1', socket.AF_INET), addr_tuple('ip2', socket.AF_INET6), addr_tuple('ip3', socket.AF_INET6)],
+            True,
+            3,
+            id='Dual IPv4/IPv6: string hostname, IPv4 resolution, multiple_ips True',
+        ),
+        pytest.param(
+            'another-hostname',
+            common.DUAL_STACK_GETADDRINFO_IPV6,
+            [addr_tuple('ip1', socket.AF_INET6), addr_tuple('ip2', socket.AF_INET), addr_tuple('ip3', socket.AF_INET6)],
+            True,
+            3,
+            id='Dual IPv4/IPv6: string hostname, IPv6 resolution, multiple_ips True',
+        ),
+        pytest.param(
+            'localhost',
+            common.SINGLE_STACK_GETADDRINFO_LOCALHOST_IPV4,
+            [addr_tuple('127.0.0.1', socket.AF_INET)],
+            False,
+            1,
+            id='Single stack IPv4: localhost, IPv4 resolution, multiple_ips False',
+        ),
+        pytest.param(
+            'localhost',
+            common.SINGLE_STACK_GETADDRINFO_LOCALHOST_IPV4,
+            [addr_tuple('127.0.0.1', socket.AF_INET), addr_tuple('ip2', socket.AF_INET)],
+            True,
+            2,
+            id='Single stack IPv4: localhost, IPv4 resolution, multiple_ips True',
+        ),
+        pytest.param(
+            'another-hostname',
+            common.SINGLE_STACK_GETADDRINFO_IPV4,
+            [addr_tuple('ip1', socket.AF_INET)],
+            False,
+            1,
+            id='Single stack IPv4: string hostname, IPv4 resolution, multiple_ips False',
+        ),
+        pytest.param(
+            'another-hostname',
+            common.SINGLE_STACK_GETADDRINFO_IPV4,
+            [addr_tuple('ip1', socket.AF_INET), addr_tuple('ip2', socket.AF_INET), addr_tuple('ip3', socket.AF_INET)],
+            True,
+            3,
+            id='Single stack IPv4: string hostname, IPv4 resolution, multiple_ips True',
+        ),
+    ],
+)
+def test_hostname_resolution(
+    aggregator, monkeypatch, hostname, getaddrinfo, expected_addrs, multiple_ips, service_checks_count
+):
+    """
+    Test that string hostnames get resolved to ipv4 address format properly.
+    """
+    instance = deepcopy(common.INSTANCE)
+    instance['host'] = hostname
+    instance['multiple_ips'] = multiple_ips
+    check = TCPCheck(common.CHECK_NAME, {}, [instance])
+
+    monkeypatch.setattr('socket.getaddrinfo', mock.Mock(return_value=getaddrinfo))
+    monkeypatch.setattr(check, 'connect', mock.Mock(side_effect=[None, None, None]))
```

Comment:
I think this is equivalent to leaving the `side_effect` out entirely.
```suggestion
    monkeypatch.setattr(check, 'connect', mock.Mock()
```